.github/workflows/pr.yml

```diff
@@ -6,11 +6,44 @@ on:
       - staging
 
 jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Cache ivy packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+          key: ${{ runner.os }}-ivy2-${{ hashFiles('**/*.sbt') }}
+          restore-keys: ${{ runner.os }}-sbt
+      - uses: olafurpg/setup-scala@v10
+      - name: Build
+        shell: bash
+        run: ./ciRunSbt.sh clean compile
+      - name: Tar artifacts
+        shell: bash
+        run:  find . -wholename "**/target/**/*" -printf '%P\0' | tar --null -C '.' --files-from=- -czf 'target.tgz'
+      - name: Store target
+        uses: actions/upload-artifact@v2
+        with:
+          name: build-target
+          path: target.tgz
   tests:
     name: Tests
     runs-on: ubuntu-latest
+    needs: [build]
     steps:
       - uses: actions/checkout@v1
+      - uses: olafurpg/setup-scala@v10
```

Comment:
Hmm, I wonder can we move `olafurpg/setup-scala@v10` to some var? Right now use it in a few places?

Reply:
I hope we'll manage to extract some common stuff in next PRs...